src/main/java/org/jabref/logic/exporter/ModsExporter.java

```diff
@@ -76,267 +49,346 @@ public void export(final BibDatabaseContext databaseContext, final Path file, Li
         }
 
         try {
-            ModsCollectionDefinition modsCollection = new ModsCollectionDefinition();
+            // create writer
+            IndentingXMLStreamWriter writer = createWriter(file);
+
             for (BibEntry bibEntry : entries) {
-                ModsDefinition mods = new ModsDefinition();
-                bibEntry.getCitationKey().ifPresent(citeKey -> addIdentifier(new UnknownField("citekey"), citeKey, mods));
+                bibEntry.getCitationKey().ifPresent(citeKey -> {
+                    try {
+                        addIdentifier(writer, new UnknownField("citekey"), citeKey);
+                    } catch (
+                            XMLStreamException e) {
+                        throw new RuntimeException(e); // cant throw SaveException?
```

Comment:
This looks a bit odd to me. What is the purpose?
Why not use the isPresent? Or set it to an empty string? e.g. ifPresent.OrElse(...)

Reply:
Are you asking about the purpose of using try-catch here? Or the use of `ifPresent(citekey ...` ? 

I used ifPresent because that is what the code had originally.

Reply:
Both. First question: Why do you need the cry catch? I would assume if there is no citeKey we set it to empty string or something like this and the other line does a similar check down below. So it's kind of duplicated code